Sources/Animations/KeyboardConstraint.swift

```diff
@@ -0,0 +1,69 @@
+import UIKit
+
+/// Set any NSLayoutConstraint class to this class and it moves up and down with the keyboard.
+open class KeyboardConstraint: NSLayoutConstraint {
+    /// this amount is added to the keyboard height to show the view above the keyboard.
+    public var offsetFromKeyboardHeight: CGFloat = 15
+    /// will be set on awakeFromNib to the constant value. 
+    /// Is used to move back to the position before the keyboard was shown
+    public var normalConstant: CGFloat = 0
+
+    private var notifications: [Any]?  = [Any]()
+
+    override open func awakeFromNib() {
+        //swiftlint:disable opening_brace
+        normalConstant = constant
+        let not1 = NotificationCenter.default.addObserver(forName: NSNotification.Name.UIKeyboardWillShow,
+                                               object: nil,
+                                               queue: OperationQueue.main)
+                    { [weak self] (notification) in
+                        let userInfo = notification.userInfo
+                        let frameInfo = userInfo?[UIKeyboardFrameBeginUserInfoKey]
+                        if let keyboardSize = (frameInfo as? NSValue)?.cgRectValue {
+                            let offset = self?.offsetFromKeyboardHeight != nil ? self!.offsetFromKeyboardHeight : 0
+                            self?.animate(to: (keyboardSize.height + offset), userInfo: userInfo)
+                        }
+                    }
+
+        let not2 = NotificationCenter.default.addObserver(forName: NSNotification.Name.UIKeyboardWillHide,
+                                                          object: nil,
+                                                          queue: OperationQueue.main)
+                    { [weak self] (notification) in
+                        let userInfo = notification.userInfo
+                        self?.animate(to: self?.normalConstant, userInfo: userInfo)
+                    }
+
+        notifications?.append(not1)
```

Comment:
🎈I'm not a fan of naming variables `not1` and `not2`. I prefer to name tham after what they represent... `willShowNotificationToken` and `willHideNotificationToken`. This way we know what they mean.

I would also save them as 2 properties on the instance instead of in an array.

My 5 💵...